### DISCUSSION TITLE
期望站点内可以支持套用字体

### DISCUSSION THREAD

**Discussion by intoyang** [halo-dev/halo#6646] [Ideas]
### 你当前使用的版本

2.5

### 描述一下此特性

1. 可在左侧导航栏「外观」集合中增加「字体」子菜单；
2. 提供官方字体库功能，内置经核验的免费可商用字体，用户可从字体库中安装需要的字体包；
3. 支持将安装好的字体一键全局启用，则站点内的全部字体都将套用当前字体包；

<img width="509" alt="image" src="https://github.com/halo-dev/halo/assets/73098982/70fcf3b1-2fb1-41c2-b97b-9878aff9fa01">

支持用户自定义字体组合，组合中可以指定多语言的指定字体。
<img width="508" alt="image" src="https://github.com/halo-dev/halo/assets/73098982/058681bb-0521-48c4-a64c-a55c3c4eae88">

字体规则可以定义站点内不同位置的字体样式，支持引用独立字体包或字体组合；初期阶段，可以使用「省心模式」，将规则范围限定在站点标题、站点说明、菜单、文章标题、文章正文几个部分；后续可以扩展出更广泛的「随心模式」，分离更细粒度的自定义配置。
<img width="505" alt="image" src="https://github.com/halo-dev/halo/assets/73098982/2efc336c-38d0-40b4-ae1e-750bdb604cb3">

**Comment by ruibaby** [comment]
似乎很适合通过插件来实现。

/kind feature
/area plugin

**Comment by ileonspace** [comment]
这个不错👍
